src/coreclr/inc/corinfo.h

```diff
@@ -1970,6 +1931,23 @@ struct CORINFO_SWIFT_LOWERING
     size_t numLoweredElements;
 };
 
+#define MAX_FPSTRUCT_LOWERED_ELEMENTS 2
+
+// Lowering information on fields of a struct passed by hardware floating-point calling convention on RISC-V and LoongArch
+// NOTE: all array elements past numLoweredElements are zero-initialized
+struct CORINFO_FPSTRUCT_LOWERING
+{
+    // Whether the struct should be passed by integer calling convention (cannot be passed by FP calling convention).
+    // If true, all other fields of CORINFO_FPSTRUCT_LOWERING are zeroed.
+    bool byIntegerCallConv;
```

Comment:
For my edification, what's the calling convention of `struct { object o; float a; }`?

Is it passed using this special lowering or using integer call conv?

Reply:
According to floating-point calling convention (special lowering). Object is a pointer, which for the convention is simply an integer field.

Reply:
Should the Object field from my example be reported as `CORINFO_TYPE_CLASS` in this struct then? It does not look right that it is reported as int.

Reply:
I think [we agreed](https://github.com/dotnet/runtime/pull/103945#discussion_r1664673204) that JIT doesn't need to know IntKind (it was used to tell ordinary integers from GC refs), because it can probe for GC refs from the layout it knows already.

Reply:
Currently JIT only needs to know from CORINFO_FPSTRUCT_LOWERING whether the type is floating or integer and the type size. GC info is taken from class layout.

I'd like to merge as is if it's not a problem, this PR blocks #104237 and subsequent changes stemming from #101796. If it turns out that the integer type needs to be more precisely specified for some reason, we can always amend it and FpStructInRegistersInfo.

Reply:
Ping. Can this be merged?

Reply:
Could you please add a comment that the types returned in `loweredElements` only represent the size and not the actual type?

(An alternative would be to change `loweredElements` to `uint32_t loweredElementsSizes[]` to make it obvious that the API just returns the size and not the actual type.)

Reply:
Added comment. The alternative would need another bool array for floating vs. integer and would stray from CORINFO_SWIFT_LOWERING if the need to handle ABI lowering in shared code ever comes up.